ultralytics_yolov5.md

```diff
@@ -0,0 +1,180 @@
+        for c in pred[:, -1].unique():
+            n = (pred[:, -1] == c).sum()  # detections per class
+            str += '%g %ss, ' % (n, model.names[int(c)])  # add to string
+        for *box, conf, cls in pred:  # xyxy, confidence, class
+            label = model.names[int(cls)] if hasattr(model, 'names') else 'class_%g' % cls
+            # str += '%s %.2f, ' % (label, conf)  # label
+            ImageDraw.Draw(img).rectangle(box, width=3)  # plot
+    img.save('results%g.jpg' % i)  # save
+    print(str + 'Done.')
+```
+
+<img src="https://user-images.githubusercontent.com/26833433/96368758-114c3a00-1156-11eb-9ede-a03f9aff42e4.jpg" width="550">
+
+<img src="https://user-images.githubusercontent.com/26833433/97121810-54dc1080-1721-11eb-8c25-451ccb1452e4.jpg" width="350">
+
+For all inference options see YOLOv5 [`autoShape()` forward method](https://github.com/ultralytics/yolov5/blob/453acdec67202e15b772370d31eb196d844f2892/models/common.py#L129-L136):
+```python
+# Example inputs for prediction=model(x)
+#   opencv:     x = cv2.imread('image.jpg')[:,:,::-1]  # HWC BGR to RGB x(720,1280,3)
+#   PIL:        x = Image.open('image.jpg')  # HWC x(720,1280,3)
+#   numpy:      x = np.zeros((720,1280,3))  # HWC
+#   torch:      x = torch.zeros(16,3,720,1280)  # BCHW
+#   multiple:   x = [Image.open('image1.jpg'), Image.open('image2.jpg'), ...]  # list of images
+```
+
+### Vary Input Channels
+To load a pretrained YOLOv5s model with 4 input channels rather than the default 3:
+```python
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True, channels=4)
+```
+In this case the model will be composed of pretrained weights **except for** the very first input layer, which is no longer the same shape as the pretrained input layer. The input layer will remain initialized by random weights.
+
+### Vary Number of Classes
+To load a pretrained YOLOv5s model with 10 output classes rather than the default 80:
+```python
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True, classes=10)
+```
+In this case the model will be composed of pretrained weights **except for** the output layers, which are no longer the same shape as the pretrained output layers. The output layers will remain initialized by random weights.
+
+
+## Model Description
+
+<img width="800" alt="YOLOv5 Models" src="https://user-images.githubusercontent.com/26833433/97808084-edfcb100-1c64-11eb-83eb-ffed43a0859f.png">
+
+YOLOv5 is an architecture for compound-scaled object detection models trained on the COCO 2017 dataset, and includes built-in functionality for Test Time Augmentation (TTA), Model Ensembling, Rectangular Inference, Hyperparameter Evolution and export to a variety of common formats such as ONNX, CoreML and TFLite. YOLOv5 also features a free **iOS App** that demos real-time 30 FPS inference of all models on iPhones and iPads: [iDetection iOS App](https://apps.apple.com/us/app/idetection/id1452689527#?platform=iphone).
```

Comment:
you can summarize this description in a shorter form right before the `Load YOLOv5 From PyTorch Hub` section

Reply:
Yes, it's been shortened now.